lldb/tools/lldb-dap/JSONUtils.cpp

```diff
@@ -1519,11 +1480,11 @@ CreateRunInTerminalReverseRequest(const llvm::json::Object &launch_request,
 // Keep all the top level items from the statistics dump, except for the
 // "modules" array. It can be huge and cause delay
 // Array and dictionary value will return as <key, JSON string> pairs
-void FilterAndGetValueForKey(const lldb::SBStructuredData data, const char *key,
-                             llvm::json::Object &out) {
-  lldb::SBStructuredData value = data.GetValueForKey(key);
+void FilterAndGetValueForKey(const lldb::SBStructuredData data,
+                             llvm::StringRef key, llvm::json::Object &out) {
+  lldb::SBStructuredData value = data.GetValueForKey(key.data());
```

Comment:
I don't think this change is good as it hides (from the caller) the fact that the function expects the argument to be null terminated. If someone passes a non-null terminated stringref, things will get.. interesting.

You can either keep the C pointer, or do the `.str().c_str()` dance I've seen in some other places.

If, in the first case, you want to get rid of the C strcmp, you can always construct a StringRef locally (`StringRef(key) == "modules"`)